### PR TITLE
[skip ci] Add delayed best practices

### PR DESCRIPTION
**Pull request by mrocklin**
- [ ] Tests added / passed
- [ ] Passes `flake8 dask`

### CODE REVIEW

**Comment by mrocklin** [comment]
This came out of saying the same things on stack overflow repeatedly

File: docs/source/delayed-best-practices.rst [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+
+
+Avoid calling delayed within delayed functions
+----------------------------------------------
+
+Often if you are new to using Dask.delayed you place dask.delayed calls
+everywhere and hope for the best.  While this may actually work it's usually
+slow and results in hard-to-understand solutions.
+
+Usually you only never call dask.delayed within dask.delayed functions.
```
[the remark this answers is not in the document]
Comment (mrocklin) [reply docs/source/delayed-best-practices.rst]: Thanks, fixed!

File: docs/source/delayed-best-practices.rst [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+
+   dask.compute(f(filenames))  # this has many tasks and so will parallelize
+
+
+Avoid too many tasks
+--------------------
+
+Every delayed task has an overhead of a few hundred microseconds.  Usually this
+is ok, but it can become a problem if you apply dask.delayed too finely.  In
+this case it's often best to break up your many tasks into batches, or use on
```
[the remark this answers is not in the document]
Comment (mrocklin) [reply docs/source/delayed-best-practices.rst]: Thanks, fixed!

**Comment by mrocklin** [comment]
Thanks for the comment @jrmlhermitte .  Is it posible that you're thinking about dask array rather than dask delayed?

**Comment by jrmlhermitte** [comment]
Yes that's my mistake sorry!
This is a great addition to the docs!

**Pull request merged**